replicator/active_replicator_pull.go

```diff
@@ -0,0 +1,71 @@
+package replicator
+
+import (
+	"fmt"
+
+	"github.com/couchbase/go-blip"
+)
+
+// ActivePullReplicator is a unidirectional pull active replicator.
+type ActivePullReplicator struct {
+	config      *ActiveReplicatorConfig
+	blipContext *blip.Context
+	blipSender  *blip.Sender
+}
+
+// Compile-time interface check.
+var _ ActiveReplicator = &ActivePullReplicator{}
+
+func NewPullReplicator(config *ActiveReplicatorConfig) *ActivePullReplicator {
+	return &ActivePullReplicator{
+		config:      config,
+		blipContext: blip.NewContextCustomID(config.ID+"-pull", blipCBMobileReplication),
+	}
+}
+
+// Connects to the target via BLIP and sets.
+func (apr *ActivePullReplicator) Connect() error {
+	if apr == nil {
+		// noop
+		return nil
```

Comment:
Is there a reason to no-op instead of error if we call Connect() on a nil replicator?

Reply:
No reason, safer to error. Changed.